src/DDCEDViewer.cc

```diff
@@ -649,19 +654,22 @@ void DDCEDViewer::drawMCParticle(dd4hep::Detector& theDetector, int& layer, unsi
     for(int i=0; i<col->getNumberOfElements() ; i++){
         MCParticle* mcp = dynamic_cast<MCParticle*> ( col->getElementAt( i ) ) ;
         float charge = mcp->getCharge ();
-        if( mcp-> getGeneratorStatus() != 1 && _usingParticleGun == false ) 
-            continue ; // stable particles only
-        if ( mcp->getEnergy() < _mcpECut ) 
+        if( mcp-> getGeneratorStatus() != 1 && _usingParticleGun == false ) {
+            if (! (_drawMCParticlesCreatedInSimulation && mcp->isCreatedInSimulation() )) {
```

Comment:
```suggestion
        if( mcp-> getGeneratorStatus() != 1 && 
             _usingParticleGun == false 
             ! (_drawMCParticlesCreatedInSimulation && mcp->isCreatedInSimulation() ) {
```
(plus removing additional braces below). Unless, I am missing something here, this should work, no?